test/createAzureResource.test.ts

```diff
@@ -59,26 +58,71 @@ suite('Create Azure Resources', async function (this: ISuiteCallbackContext): Pr
         ext.azureAccountTreeItem.dispose();
     });
 
-    test('createFunctionApp (Basic)', async () => {
+    test('Create windows function app (Basic) and deploy JavaScript project', async () => {
         resourceGroupsToDelete.push(resourceName1);
-        await runWithFuncSetting('advancedCreation', undefined, async () => {
-            ext.ui = new TestUserInput([resourceName1]);
-            await vscode.commands.executeCommand('azureFunctions.createFunctionApp');
-            const createdApp: WebSiteManagementModels.Site = await webSiteClient.webApps.get(resourceName1, resourceName1);
-            assert.ok(createdApp);
+        await runWithFuncSetting('projectLanguage', ProjectLanguage.JavaScript, async () => {
+            await runWithFuncSetting('projectRuntime', ProjectRuntime.v2, async () => {
+                const functionName: string = 'HttpTrigger';
+                const templateId: string = 'HTTP trigger';
+                const authLevel: string = 'Function';
+                ext.ui = new TestUserInput([projectPath, templateId, functionName, authLevel]);
+                await vscode.commands.executeCommand('azureFunctions.createNewProject');
+                await validateProject(projectPath, getJavaScriptValidateOptions(true));
+                await runWithFuncSetting('advancedCreation', undefined, async () => {
+                    ext.ui = new TestUserInput(['$(plus) Create New Function App in Azure', resourceName1]);
+                    await vscode.commands.executeCommand('azureFunctions.deploy');
+                    await delay(500);
+                    // Verify the deployment result through copyFunctionUrl
+                    await vscode.env.clipboard.writeText(''); // Clear the clipboard
+                    ext.ui = new TestUserInput([resourceName1, functionName]);
+                    await vscode.commands.executeCommand('azureFunctions.copyFunctionUrl');
+                    const functionUrl: string = await vscode.env.clipboard.readText();
+                    const result: string = await getBody(functionUrl, resourceName1);
+                    assert.equal(result, `Hello ${resourceName1}`, `The result should be "Hello ${resourceName1}" rather than ${result} and the triggerUrl is ${functionUrl}`);
+                });
+            });
+        });
+    });
+
+    test('Deploy CSharp project (windows)', async () => {
+        const resourceName2: string = getRandomHexString().toLowerCase();
```

Comment:
This test and the javascript test have a lot of duplicate code. Can you create a function that looks similar to this:
```
async function testCreateProjectAndDeploy(createProjectInputs: string[], projectValidateOptions: IValidateProjectOptions, <any other parameters>): Promise<void> {
```
and then call that function from each test.